Lesson16/lesson16/gokce/script.js

```diff
@@ -0,0 +1,388 @@
+  let reversedSentence = [];
+  reversedSentence = sentence.split(" ").reverse().join(" ");
+  console.log(reversedSentence);
+}
+
+reverseWords("Life is challenging sometimes.");
+/*
+15. Filter Words Longer Than X
+    - Define a function `filterLongWords(words, minLength)` that uses a for loop
+      to collect only the words that have a length >= minLength.
+    - Log the resulting array.
+*/
+function filterLongWords(words, minLength) {
+  let resultArray = [];
+  for (let i = 0; i < words.length; i++) {
+    if (words[i].length >= minLength) {
+      resultArray.push(words[i]);
+    }
+  }
+  console.log(resultArray);
+}
+filterLongWords(["apple", "banana", "cherry", "kiwi"], 6);
+
+/*
+16. Log Array Elements with Their Indices
+    - Define a function `logElementsWithIndex(arr)` that loops through the array
+      and logs "Index: i, Value: arr[i]" for each element.
+*/
+
+function logElementsWithIndex(arr) {
+  for (const element of arr) {
+    console.log(`Index ${element},Value ${arr[element]}`);
+  }
+}
+
+function logElementsWithIndex2(arr) {
+  let index = 0;
+  for (const element in arr) {
+    console.log(`Index ${index},Value ${arr[element]}`);
+    index++;
+  }
+}
+
+logElementsWithIndex2(["apple", "banana", "cherry", "kiwi"]);
+
+/*
+17. Find the Smallest Number in an Array
+    - Define a function `findMin(numbers)` that loops through the array
+      to find and return the smallest number.
+    - Log the smallest number.
+*/
+function findMin(numbers) {
+  let smallestNum = numbers[0];
+  for (let i = 0; i < numbers.length; i++) {
+    if (numbers[i] < smallestNum) {
+      smallestNum = numbers[i];
+    }
+  }
+
+  return smallestNum;
+}
+
+findMin([2,4,6,7,34,1,-1,-4]);
+
+function findMin2(numbers) {
+  let minNum = numbers[0];
+  let index = 1;
+  do {
+    if (numbers[index] < minNum) {
+      minNum = numbers[index];
+    }
+    index++;
+  } while (index < numbers.length);
+  console.log(minNum);
+  return minNum;
+}
+findMin2([2,4,6,7,34,1,-1,-4]);
+/*
+18. Count Occurrences of a Word in an Array
+    - Define a function `countOccurrences(arr, word)` that loops through `arr`
+      to count how many times `word` appears.
+    - Log the count.
+*/
+
+let exampleArray = ["apple", "banana", "apple", "orange", "apple"];
+let wordToCount = "apple";
+
+function countOccurrences(arr, word) {
+  let occurrencesNum = 0;
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] === word) {
+      ++occurrencesNum;
+    }
+  }
+  return occurrencesNum;
+}
+
+countOccurrences(exampleArray,wordToCount)
+
+
+
+/*
+19. Remove Falsy Values
+    - Define a function `removeFalsyValues(arr)` that loops through an array
+      and returns a new array without falsy values (false, 0, "", null, undefined, NaN).
+    - Log the new array.
+*/
+
+function removeFalsyValues(arr) {
+  let withoutFalsyArr = [];
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i]) {
+      withoutFalsyArr = withoutFalsyArr.push(arr[i]);
```

Comment:
Push method modifies the array directly therefore you don't need to assign `withoutFalsyArr` value to the result of the operation. 
See more here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push
```suggestion
      withoutFalsyArr.push(arr[i]);
```